numpy/lib/function_base.py

```diff
@@ -91,6 +101,17 @@ def _hist_optim_numbins_estimator(a, estimator):
     if a.size == 0:
         return 1
 
+    if data_weights is not None:
+        raise TypeError("Automated estimation of the number of "
```

Comment:
How about a `RuntimeWarning`? Something along the lines of `FD estimator currently ignores weighted data`. Might be better than crashing.